plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/BridgingHiveMetastore.java

```diff
@@ -154,6 +158,52 @@ public List<String> getAllViews(String databaseName)
         return delegate.getAllViews(databaseName);
     }
 
+    @Override
+    public Map<String, List<String>> getAllTables()
+    {
+        checkState(supportBatchListingOperations(), "Batch listing operations not supported");
+        return delegate.getAllTables();
+    }
+
+    @Override
+    public Map<String, List<String>> getAllViews()
+    {
+        checkState(supportBatchListingOperations(), "Batch listing operations not supported");
+        return delegate.getAllViews();
+    }
+
+    @Override
+    public boolean supportBatchListingOperations()
+    {
+        return switch (batchListingOperationsSupport) {
+            case UNKNOWN -> {
+                if (!delegate.supportBatchGetViews()) {
+                    batchListingOperationsSupport = BatchListingOperationsSupport.UNSUPPORTED;
+                }
+                else {
+                    batchListingOperationsSupport = checkBatchListingOperationsSupport();
```

Comment:
Why not just rely on `delegate.supportBatchGetViews()` being correct?
It's weird to have to ask, and then check the answer for correctness. Classes should have more trust than that.

`BridgingHiveMetastore`'s metastore logic doesn't need to deal with different HMS versions. It seems to me the check is applied on a wrong level

Reply:
`delegate.supportBatchGetViews()` simply checks whether `translate-hive-views` is enabled.
The `checkBatchListingOperationsSupport` checks whether the metastore actually supports the batch thing.

It is weird, I agree. I'll take in into consideration while rewriting this to facilitate your other comments

Reply:
> `delegate.supportBatchGetViews()` simply checks whether `translate-hive-views` is enabled.

That's the current implementation.

> The checkBatchListingOperationsSupport checks whether the metastore actually supports the batch thing.

That precisely describes what the code here is doing.

Why not have delegate do both?

Reply:
The delegate does not know whether all tables are needed or just a single one, but we ask for all since the call is cheap.
Let's take simple scenario:
1.Some external entity asks for tables for schema A.
2.Caching metasore redirects to delegate.getAllTables.
3.The delegate does not support batch methods so it iterates over schemas and fetches tables the old way, which takes a lot of time.
So it's a matter of caching metasore knowing whether it can invoke getAllTables or getables on the delegate.
One solution for this is the #1 from https://github.com/trinodb/trino/pull/16733#discussion_r1152825790. We explicitly add a getAllTables method and it is invoked only when all tables are indeed needed. Other use cases remain unchanged.
WDYT?

Reply:
i understand now why Caching metasore  may have mor knowledge, but I don't follow why `BridgingHiveMetastore.java` woiuld have more knowledge than the underlying impl

are  we talking in context of bridging or caching?